### PR TITLE
SPECIES NUKING 2023: Moves speed modifier to bodyparts

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

As we all know, speed is stored in the legs.

## Why It's Good For The Game

Potential for more medical abominations.
Potential for adding augmented legs that make you speedier (?)

## Changelog

:cl:
refactor: Species speed is now stored in bodyparts. Leg transplants from slower species will make you slower.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/zombies.dm [hunk 1]
```diff
@@ -120,6 +119,17 @@
...
 	)
 
+	// Infectious zombies have slow legs
+	bodypart_overrides = list(
+		BODY_ZONE_HEAD = /obj/item/bodypart/head/zombie,
+		BODY_ZONE_CHEST = /obj/item/bodypart/chest/zombie,
+		BODY_ZONE_L_ARM = /obj/item/bodypart/arm/left/zombie,
+		BODY_ZONE_R_ARM = /obj/item/bodypart/arm/right/zombie,
+		BODY_ZONE_L_LEG = /obj/item/bodypart/leg/left/zombie/infectious,
+		BODY_ZONE_R_LEG = /obj/item/bodypart/leg/right/zombie/infectious
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/species_types/zombies.dm:129]:
```suggestion
		BODY_ZONE_R_LEG = /obj/item/bodypart/leg/right/zombie/infectious,
```

**Review by san7890 — CHANGES_REQUESTED** [review]
i would rather you not include a changelog at all for this PR unless changing the way speed is calculated for things like zombies/snailpeople is changed in a way where it's noticeably slower or faster.

if it's approximately the same (or as close as you can possibly manage here) maybe something like "the way speed is calculated for snailpeople/zombies/etc. has been updated slightly, expect to see some changes" would work for a changelog since that is appropriate information to be player-facing

**Comment by ChungusGamer666** [comment]
@san7890 updated the changelog, i think this is pretty player facing since now getting legs from a snail for example will impact your speed as if you were one

**Comment by ChungusGamer666** [comment]
but like it could go either way, i can remove the changelog if this is still unnecessary in your opinion

**Comment by san7890** [comment]
that's better thanks

File: code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm [hunk 2]
```diff
@@ -377,13 +389,15 @@
...
 	burn_modifier = 1.25
+	speed_modifier = 0.75 //big big fungus
 
 /obj/item/bodypart/leg/right/mushroom
 	limb_id = SPECIES_MUSHROOM
 	unarmed_damage_low = 9
 	unarmed_damage_high = 21
 	unarmed_stun_threshold = 14
 	burn_modifier = 1.25
+	speed_modifier = 0.75 //big big fungus
```
Comment (ATH1909) [inline code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm:400]:
```suggestion
	speed_modifier = 0.75 //big fungus big fungus
```

**Comment by ATH1909** [comment]
this seems like it could have major powergaming repercussions

e.g. getting your legs swapped out as a snail to become turbo

**Comment by ChungusGamer666** [comment]
fucking wack ass integration tests

**Comment by ChungusGamer666** [comment]
> this seems like it could have major powergaming repercussions
> 
> e.g. getting your legs augged as a snail to become turbo

I mean, that is kind of the point, letting technology fix your terrible terrible inborn deficiencies

**Comment by Time-Green** [comment]
We might have to rethink this later if we get speed species

**Pull request merged**